src/CommandLineUtils/HelpText/DescriptionFormatter.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+
+namespace McMaster.Extensions.CommandLineUtils.HelpText
+{
+    /// <summary>
+    /// A formatter for creating nicely wrapped descriptions for display on the command line in the second column
+    /// of generated help text.
+    /// </summary>
+    public class DescriptionFormatter
+    {
+        /// <summary>
+        /// The default console width used for wrapping if the width cannot be gotten from the Console.
+        /// </summary>
+        public const int DefaultConsoleWidth = 80;
+        private int _paddedLineLength;
+        private int _consoleWidth;
+        private string _paddedLine;
+
+        /// <summary>
+        /// A description formatter for dynamically wrapping the description to print in a CLI usage.
+        /// </summary>
+        /// <param name="firstColumnWidth">The width of the first column where the option/argument/command is listed.</param>
+        /// <param name="spacerLength">The number of spaces separating the columns.</param>
+        /// <param name="consoleWidth">The max width of the console controlling when to wrap to a new line.
+        /// Defaults to Console.BufferWidth.
+        /// </param>
+        public DescriptionFormatter(int firstColumnWidth, int spacerLength, int? consoleWidth = null)
+        {
+            _paddedLineLength = firstColumnWidth + spacerLength;
+            _consoleWidth = consoleWidth ?? GetDefaultConsoleWidth();
+            _paddedLine = Environment.NewLine + new string(' ', _paddedLineLength);
+        }
+
+        /// <summary>
+        /// Get the Console width.
+        /// </summary>
+        /// <returns>BufferWidth or the default.</returns>
+        private int GetDefaultConsoleWidth()
+        {
+            try
+            {
+                return Console.BufferWidth;
+            }
+            catch (System.IO.IOException)
+            {
+                // If there isn't a console - for instance in test enviornments
+                // An IOException will be thrown trying to get the Console.BufferWidth.
+                return DefaultConsoleWidth;
+            }
+        }
+
+        /// <summary>
+        /// Dynamically wrap a description.
+        /// </summary>
+        /// <param name="original">The original description text.</param>
+        /// <returns>Dynamically wrapped description with explicit newlines preserved.</returns>
+        public string Wrap(string original)
```

Comment:
Aha I was wondering what the nice way of handling this was. This is required because of linting rules correct? Normally a string can be null. What exactly is requiring explicit "allowed to be null" declarations?

Reply:
This is required because I've enabled a new C# 8.0 feature called 'non-nullable reference types'.

https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references